### PR TITLE
31 temperature and smoke detector

### PR DESCRIPTION
**Pull request by drrk**
NF31 Smoke detector

Conversion from Sparrow
Code tested (but not with smoke)
Notes send correctly, writeup complete.

### CODE REVIEW

File: 31-temperature-and-smoke-detector/README.md [hunk 1]
```diff
@@ -19,72 +22,63 @@ Temperature monitoring and smoke detection across a number of rooms at a facilit
...
+First connect your Blues Swan and Notecard to your Notecarrier-F.
 
-1. Splice a male to female jumper in half and solder the exposed wires to the `GND` pad on the 5v converter. Connect the female end to `GND` on the sensor and the male end to the `GND` pin on Sparrow.
-2. Splice another jumper, and solder the exposed wire on male half to `Vi` on the converter. Connect the male connector to `VBAT` on Sparrow. Solder the exposed wire on the female half to `Vo` and connect the female end to the `A0` pin on the MQ2 sensor.
-3. Using an intact male-to-female jumper, connect `A0` on the sensor to the `A1` pin on Sparrow.
-4. The 4th pin on the sensor is left unconnected.
+1. Follow the steps in the [Notecard Quickstart](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-f/#connect-your-notecard-and-notecarrier) to connect your Notecard LoRa to your Notecarrier-F.  Your antenna will only have one cable, and the Notecard LoRa only has one connection.
```
Comment (tjvantoll) [inline 31-temperature-and-smoke-detector/README.md:33]:
```suggestion
1. Follow the steps in the [Notecard Quickstart](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-f/#connect-your-notecard-and-notecarrier) to connect your Notecard LoRa to your Notecarrier F.  Your antenna will only have one cable, and the Notecard LoRa only has one connection.
```

File: 31-temperature-and-smoke-detector/README.md [hunk 2]
```diff
@@ -19,72 +22,63 @@ Temperature monitoring and smoke detection across a number of rooms at a facilit
...
+3. Attach the Swan to your computer with a Micro USB to USB-A cable, using the **Micro USB port on the Swan**.
 
-1. You will need to fetch the dependencies required by the firmware. After cloning this repository, run these commands: `git submodule update --init 31-temperature-and-smoke-detector/firmware/note-c` and `git submodule update --init 31-temperature-and-smoke-detector/firmware/sparrow-lora`. This will fetch the `note-c` and `sparrow-lora` submodules that this project depends on.
+![Notecard LoRa in Notecarrier F with antenna](images/notecarrier-notecard.jpg)
 
-2. Connect the STLINK-V3MINI to your development PC with a USB A to micro USB cable.
+### MQ2 Sensor Connection
 
-3. Connect the STLINK to your reference node with the 2x7 JTAG ribbon cable.
+This solution makes use of the an MQ2 sensor module, which detects smoke, as well as various gasses. Using the 3 male-to-female jumper wires, connect the MQ2 sensor to the Sparrow Reference Sensor board and the v5 converter as follows:
```
Comment (tjvantoll) [inline 31-temperature-and-smoke-detector/README.md:43]: > Sparrow Reference Sensor board

Missed this one

File: 31-temperature-and-smoke-detector/README.md (hunk 1 above)
Comment (tjvantoll) [reply 31-temperature-and-smoke-detector/README.md:33]: There are a few other “Notecarrier-F” references still in this doc so a change all to make them all “Notecarrier F”.

File: 31-temperature-and-smoke-detector/README.md [hunk 3]
```diff
@@ -19,72 +22,63 @@ Temperature monitoring and smoke detection across a number of rooms at a facilit
...
 
-4. There are a few ways to build and flash the firmware onto the reference node. These are covered in the [Sparrow Builder's Guide](https://dev.blues.io/sparrow/sparrow-builders-guide/). Follow the steps in that guide and then return to these instructions.
+1. Solder a set of headers onto the 5v converter.
+1. Connetor a jumper to `GND` pad on the 5v converter and the other end  to a `GND` pin on the Notecarrier-F.
+1. Connect a jumper to `GND` on the sensor and the other end  to a `GND` pin on the Notecarrier-F.
+2. Connect a jumper to `Vi` on the converter and the other end to  `F_3V3` on the Notecarrier-F. 
+2. Connect a jumper to `Vo` on the converter and the other end to the  `VCC` pin on the MQ2 sensor.
+3. Connect a jumper to `A0` on the sensor and the other end to the `F_A0` pin on the Notecarrier-F.
+4. The 4th pin on the sensor is left unconnected.
```
Comment (tjvantoll) [inline 31-temperature-and-smoke-detector/README.md:51]: You can start all of these with `1.`

Currently you have 1, 1, 2, 2, 3, 4 :)

File: 31-temperature-and-smoke-detector/README.md [hunk 4]
```diff
@@ -19,72 +22,63 @@ Temperature monitoring and smoke detection across a number of rooms at a facilit
...
 
-```
-mq2: note sent. gas: 4226, temperature: 20.2C, humidity: 35.26%
-```
+The firmware provided uses the Arduino Framework, follow the [instructions in the git repo](firmware/) to flash your Swan.
 
 ## Temperature Monitoring and Smoke Alerting Behavior
 
-The firmware periodically monitors temperature and gas concentrations, and posts events with this information. Regular monitoring events are not synced immediately to Notehub, and will be delivered as often as the Notecard is configured to sync with Notehub via the [`hub.set`](https://dev.blues.io/api-reference/notecard-api/hub-requests/#hub-set) request.
+The firmware periodically monitors temperature and gas concentrations, and posts events with this information. Regular monitoring events are not synced immediately to Notehub, and will be delivered as often as the Notecard is configured to sync with Notehub via the [`hub.set`](https://dev.blues.io/api-reference/notecard-api/hub-requests/#hub-set) request.  This value is configure with the `MQ2_SYNC_PERIOD` define in the file [`./firmware/main.cpp`](./firmware/main.cpp)
```
Comment (tjvantoll) [inline 31-temperature-and-smoke-detector/README.md:71]:
```suggestion
The firmware periodically monitors temperature and gas concentrations, and posts events with this information. Regular monitoring events are not synced immediately to Notehub, and will be delivered as often as the Notecard is configured to sync with Notehub via the [`hub.set`](https://dev.blues.io/api-reference/notecard-api/hub-requests/#hub-set) request.  You can configure this value with the `MQ2_SYNC_PERIOD` defined in the file [`./firmware/main.cpp`](./firmware/main.cpp) 
```

File: 31-temperature-and-smoke-detector/firmware/src/main.cpp [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+static bool isAlert(MQ2SensorReading& reading);
+static bool registerNotefileTemplate();
+static void addNote(const MQ2SensorReading& reading, bool immediate, uint8_t alert);
+
+
+Notecard notecard;
+BME280 bmeSensor;
+// Uncomment this line and replace com.your-company:your-product-name with your
+// ProductUID.
+#define PRODUCT_UID "net.bowerham.kimball:temperaturesmokedetector"
```
Comment (tjvantoll) [inline 31-temperature-and-smoke-detector/firmware/src/main.cpp:37]: You left your productuid in here :)

**Review by tjvantoll — APPROVED** [review]
Awesome stuff again! Only had a few notes. Thanks again for all your help on these @drrk.

File: 31-temperature-and-smoke-detector/README.md (hunk 3 above)
Comment (drrk) [reply 31-temperature-and-smoke-detector/README.md:51]: That's what happens when you decide to make the instructions make more sense and add steps!! I'll change them all :)

File: 31-temperature-and-smoke-detector/firmware/src/main.cpp (hunk 5 above)
Comment (drrk) [reply 31-temperature-and-smoke-detector/firmware/src/main.cpp:37]: Well it was bound to happen eventually.  I had better delete that project!  Will push the fix too

**Pull request merged**